lib/resty/upstream/healthcheck.lua

```diff
@@ -435,6 +474,49 @@ local function get_lock(ctx)
     return true
 end
 
+local function set_all_down_flag(ctx, ppeers, bpeers)
```

Comment:
Please add more comment to this function since I have difficulties understanding what this function tries to do. It definitely does WAY more magic than simply setting an "all down" flag.

Reply:
Added brief description of the function.